probe-rs/src/probe/daplink/tools.rs

```diff
@@ -144,6 +144,28 @@ pub fn open_v2_device(device: Device<rusb::Context>) -> Option<DAPLinkDevice> {
     None
 }
 
+fn device_matches(
+    device_descriptor: DeviceDescriptor,
+    selector: &DebugProbeSelector,
+    serial_str: &str,
+) -> bool {
+    if device_descriptor.vendor_id() == selector.vendor_id
+        && device_descriptor.product_id() == selector.product_id
+    {
+        if let Some(serial) = selector.serial_number {
+            if serial == serial_str {
+                true
+            } else {
+                false
+            }
+        } else {
+            false
```

Comment:
Did you mean for this one to be `true` so it matches when selector.serial_number is None?

Reply:
Yes, thanks :)